api/nodes/serializers.py

```diff
@@ -35,6 +35,45 @@
 from osf.utils import permissions as osf_permissions
 
 
+def get_institutions_to_add_remove(institutions, new_institutions):
+    diff = relationship_diff(
+        current_items={inst._id: inst for inst in institutions.all()},
+        new_items={inst['_id']: inst for inst in new_institutions}
+    )
+
+    insts_to_add = []
+    for inst_id in diff['add']:
+        inst = Institution.load(inst_id)
+        if not inst:
+            raise exceptions.NotFound(detail='Institution with id "{}" was not found'.format(inst_id))
+        insts_to_add.append(inst)
+
+    return insts_to_add, diff['remove'].values()
+
+
+def update_institutions(node, new_institutions, user, post=False):
+    add, remove = get_institutions_to_add_remove(
+        institutions=node.affiliated_institutions,
+        new_institutions=new_institutions
+    )
+
+    if post and not len(add):
+        raise RelationshipPostMakesNoChanges
+
+    if not post:
+        for inst in remove:
+            if not user.is_affiliated_with_institution(inst) and not node.has_permission(user, 'admin'):
+                raise exceptions.PermissionDenied(
```

Comment:
This looks like something that should go on the view's permission class.

Reply:
On the node view's permissions class? This isn't going through the node-institutions view, so I'm not sure that's correct in this case.